cmd/thv-operator/controllers/mcpregistry_controller.go

```diff
@@ -211,16 +392,443 @@ func (r *MCPRegistryReconciler) finalizeMCPRegistry(ctx context.Context, registr
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+					Annotations: map[string]string{
+						"toolhive.stacklok.dev/config-hash": r.getSourceDataHash(mcpRegistry, sourceHandler),
+					},
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: "toolhive-registry-api",
+					Containers: []corev1.Container{
+						{
+							Name:  registryAPIContainerName,
+							Image: getRegistryAPIImage(),
+							Args: []string{
+								"serve",
+							},
+							Ports: []corev1.ContainerPort{
+								{
+									ContainerPort: 8080,
+									Name:          "http",
+									Protocol:      corev1.ProtocolTCP,
+								},
+							},
+							// Add resource limits and requests for production readiness
+							Resources: corev1.ResourceRequirements{
+								Requests: corev1.ResourceList{
+									corev1.ResourceCPU:    resource.MustParse("100m"),
+									corev1.ResourceMemory: resource.MustParse("128Mi"),
+								},
+								Limits: corev1.ResourceList{
+									corev1.ResourceCPU:    resource.MustParse("500m"),
+									corev1.ResourceMemory: resource.MustParse("512Mi"),
+								},
+							},
+							// Add liveness and readiness probes
+							LivenessProbe: &corev1.Probe{
+								ProbeHandler: corev1.ProbeHandler{
+									HTTPGet: &corev1.HTTPGetAction{
+										Path: "/health",
+										Port: intstr.FromInt32(8080),
+									},
+								},
+								InitialDelaySeconds: 30,
+								PeriodSeconds:       10,
+							},
+							ReadinessProbe: &corev1.Probe{
+								ProbeHandler: corev1.ProbeHandler{
+									HTTPGet: &corev1.HTTPGetAction{
+										Path: "/readiness",
+										Port: intstr.FromInt32(8080),
+									},
+								},
+								InitialDelaySeconds: 5,
+								PeriodSeconds:       5,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// Configure storage-specific aspects using StorageManager
+	if err := r.storageManager.ConfigureDeployment(deployment, mcpRegistry, registryAPIContainerName); err != nil {
+		return nil, fmt.Errorf("failed to configure deployment with storage manager: %w", err)
+	}
+
+	// Set owner reference for automatic garbage collection
+	if err := controllerutil.SetControllerReference(mcpRegistry, deployment, r.Scheme); err != nil {
+		return nil, fmt.Errorf("failed to set controller reference for deployment: %w", err)
+	}
+
+	return deployment, nil
+}
+
+// getSourceDataHash calculates the hash of the source ConfigMap data using the provided source handler
+// This hash is used as a deployment annotation to trigger pod restarts when data changes
+func (*MCPRegistryReconciler) getSourceDataHash(
+	mcpRegistry *mcpv1alpha1.MCPRegistry, sourceHandler sources.SourceHandler,
+) string {
+	// Get current hash from source using the existing handler
+	hash, err := sourceHandler.CurrentHash(context.Background(), mcpRegistry)
+	if err != nil {
+		// If we can't get the hash, return a time-based hash
```

Comment:
I don't think `fmt.Sprintf("error-%d", time.Now().Unix())` is a "time-based hash". Either update the comment or the code?